libs/extended-verification/src/Verification.service.ts

```diff
@@ -0,0 +1,192 @@
+import { Result, ResultObject } from '@blockchain-lab-um/utils';
+import {
+  VerifiableCredential,
+  VerifiablePresentation,
+  W3CVerifiableCredential,
+  W3CVerifiablePresentation,
+} from '@veramo/core';
+import { decodeCredentialToObject } from '@veramo/utils';
+import { normalizePresentation } from 'did-jwt-vc';
+import { Provider } from 'ethers';
+
+import { createVeramoAgent, type Agent } from './createVeramoAgent';
+
+export interface CredentialVerificationResult {
+  signature: {
+    isValid: boolean;
+    errors: string[];
+  };
+  schema: {
+    isValid: boolean;
+    errors: string[];
+  };
+  isExpired: boolean;
+  isNotYetValid: boolean;
+  isRevoked: boolean;
+  ebsiTrustedIssuer: boolean;
+}
+
+export interface PresentationVerificationResult {
+  signature: {
+    isValid: boolean;
+    errors: string[];
+  };
+  isExpired: boolean;
+  isNotYetValid: boolean;
+}
+
+export interface VerificationResult {
+  verified: boolean;
+  details: {
+    credentials: CredentialVerificationResult[];
+    presentation: PresentationVerificationResult | null;
+  };
+}
+
+export interface VerificationServiceInitOptions {
+  providers?: Record<string, Provider>;
+}
+
+export class VerificationService {
+  private static veramoAgent: Agent;
+
+  static async init(options?: VerificationServiceInitOptions): Promise<void> {
+    this.veramoAgent = await createVeramoAgent();
+  }
+
+  static async verify(
+    data: W3CVerifiablePresentation | W3CVerifiableCredential,
+    options?: { ebsiChecks?: boolean }
+  ): Promise<Result<VerificationResult>> {
+    const { ebsiChecks } = options || { ebsiChecks: false };
```

Comment:
`ebsiChecks` is not used anywhere. Is this intentional?

Reply:
This is intentional, we will use this variable when we do EBSI checks.